source/common/tcp/conn_pool.cc

```diff
@@ -55,6 +63,7 @@ void ActiveTcpClient::clearCallbacks() {
   tcp_connection_data_ = nullptr;
   parent_.onStreamClosed(*this, true);
   parent_.checkForIdleAndCloseIdleConnsIfDraining();
+  setIdleTimer();
```

Comment:
Can the previous line result in this connection getting closed? If so, maybe move this line up.

Reply:
Nice catch. Thanks!